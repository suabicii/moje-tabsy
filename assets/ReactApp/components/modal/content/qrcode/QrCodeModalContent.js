import React from "react";
import {useSelector} from "react-redux";

function QrCodeModalContent({setIsModalOpen}) {
    const darkMode = useSelector(state => state.darkMode);

    return (
        <div className="container text-center" role="dialog" data-testid="qr-code-modal">
            <h2 className="text-dark-emphasis">
                Zeskanuj poni≈ºszy kod QR w aplikacji <strong>Mediminder Alerts <i
                className="fa-solid fa-qrcode"></i></strong>
            </h2>
            <p className="lead">Miejsce na kod QR</p>
            <button
                data-testid="btn-close-modal"
                className={`btn ${darkMode ? 'btn-light' : 'btn-dark'}`}
                onClick={() => {
                    setIsModalOpen(false);
                }}
            >Zamknij <i className="fa-solid fa-close"></i>
            </button>
        </div>
    );
}

export default QrCodeModalContent;