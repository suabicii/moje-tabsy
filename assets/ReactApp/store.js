import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice"
import userSlice from "./features/user/userSlice";
import darkModeSlice from "./features/darkMode/darkModeSlice";

export default configureStore({
    reducer: {
        drugs: drugsSlice,
        user: userSlice,
        darkMode: darkModeSlice
    }
});