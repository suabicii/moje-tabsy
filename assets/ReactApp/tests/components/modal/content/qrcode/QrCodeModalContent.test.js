/**
 * @jest-environment jsdom
 * */
import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import store from "../../../../../store";
import QrCodeModalContent from "../../../../../components/modal/content/qrcode/QrCodeModalContent";
import {fireEvent, render, screen, waitFor} from "@testing-library/react";
import exampleQrCodeSrc from "../../../fixtures/exampleQrCodeSrc";

const setIsModalOpen = jest.fn();

function WrappedComponent({isQrCodeRequested}) {
    return (
        <Provider store={store}>
            <QrCodeModalContent isQrCodeRequested={isQrCodeRequested} setIsModalOpen={setIsModalOpen}/>
        </Provider>
    );
}

it('should correctly render QrCodeModalContent', () => {
    const tree = renderer.create(<WrappedComponent/>);

    expect(tree).toMatchSnapshot();
});

it('should close modal after clicking button', () => {
    render(<WrappedComponent/>);

    fireEvent.click(screen.getByTestId('btn-close-modal'));

    expect(setIsModalOpen).toBeCalled();
});

it('should get QR code', async () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
        text: () => Promise.resolve(`<img id="qr-code" src="${exampleQrCodeSrc}" alt="qr code">`)
    }));
    const {container} = render(<WrappedComponent isQrCodeRequested={true}/>);

    await waitFor(() => {
        const qrCode = container.querySelector('#qr-code');
        expect(qrCode).toBeTruthy();
    });
});

it('should display error if fetching QR code failed', async () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.reject({
        text: () => Promise.reject('Error')
    }));
    const {container} = render(<WrappedComponent isQrCodeRequested={true}/>);

    await waitFor(() => {
        const qrError = container.querySelector('#qr-error');
        expect(qrError).toBeTruthy();
    });
});
