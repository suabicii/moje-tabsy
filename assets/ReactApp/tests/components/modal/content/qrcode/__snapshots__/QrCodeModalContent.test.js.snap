// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render QrCodeModalContent 1`] = `
<div
  className="container text-center"
  data-testid="qr-code-modal"
  role="dialog"
>
  <h2
    className="text-dark-emphasis"
  >
    Zeskanuj poni≈ºszy kod QR w aplikacji 
    <strong>
      MediMinder Alerts 
      <i
        className="fa-solid fa-qrcode"
      />
    </strong>
  </h2>
  <div
    id="qr-placement"
  />
  <button
    className="btn btn-dark"
    data-testid="btn-close-modal"
    onClick={[Function]}
  >
    Zamknij 
    <i
      className="fa-solid fa-close"
    />
  </button>
</div>
`;
